xarray/core/indexing.py

```diff
@@ -1160,7 +1160,10 @@ def _indexing_array_and_key(self, key):
             key = key.tuple
         elif isinstance(key, BasicIndexer):
             array = self.array
-            key = key.tuple
+            # We're not interested in obtaining a scalar, but rather a 0d slice.
+            # This is accomplished by appending an ellipsis (see
+            # https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#detailed-notes).  # noqa
+            key = key.tuple + (Ellipsis,)
```

Comment:
Two questions:
1. does this also work in the other branches, i.e., the ones that do fancy indexing?
2. (related) can you remove the `_ensure_ndarray` method above?

Reply:
1. It does not affect outer indexing but breaks vectorized indexing, where some arrays come back with dimensions in a different order. I haven't looked into exactly why. It would be nice to move the change into `__getitem__`, as `return array[key + (Ellipsis,)]`, but I'm afraid that won't work. In any case, I think only basic indexing ever returns a 0d array, so the current change covers all relevant cases.

2. In fact, yes. I was convinced there were edge cases with object arrays etc., so I let it stay, but all tests pass without. Removed now.